src/theory/sets/cardinality_extension.cpp

```diff
@@ -57,20 +65,113 @@ void CardinalityExtension::registerTerm(Node n)
     d_eqc_to_card_term[r] = n;
     registerCardinalityTerm(n[0]);
   }
-  if (tnc.isInterpretedFinite())
+  Trace("sets-card-debug") << "...finished register term" << std::endl;
+}
+
+void CardinalityExtension::checkFiniteTypes()
+{
+  for (std::pair<const TypeNode, bool>& pair : d_t_card_enabled)
   {
-    std::stringstream ss;
-    ss << "ERROR: cannot use cardinality on sets with finite element "
-          "type (term is "
-       << n << ")." << std::endl;
-    throw LogicException(ss.str());
-    // TODO (#1123): extend approach for this case
+    TypeNode type = pair.first;
+    if (pair.second && type.isInterpretedFinite())
+    {
+      checkFiniteType(type);
+    }
+  }
+}
+
+void CardinalityExtension::checkFiniteType(TypeNode& t)
+{
+  Assert(t.isInterpretedFinite());
+  NodeManager* nm = NodeManager::currentNM();
+  // get the universe set (as univset (Set t))
+  Node univ = d_state.getUnivSet(nm->mkSetType(t));
+  std::map<Node, Node>::iterator it = d_univProxy.find(univ);
+  Node proxy;
+  if (it == d_univProxy.end())
+  {
+    // Force cvc4 to build the cardinality graph for the universe set
+    proxy = d_state.getProxy(univ);
+    d_univProxy[univ] = proxy;
+  }
+  else
+  {
+    proxy = it->second;
+  }
+
+  // get all equivalent classes of type t
+  vector<Node> representatives = d_state.getSetsEqClasses(t);
+  // get the cardinality of the finite type t
+  Cardinality card = t.getCardinality();
+  if (!card.isFinite())
+  {
+    // ToDo (#1123): support uninterpreted sorts with --finite-model-find
+    std::stringstream message;
+    message << "The cardinality " << card << " of the finite type " << t
+            << " is not supported yet." << endl;
+    Assert(false) << message.str().c_str();
+  }
+
+  Node typeCardinality = nm->mkConst(Rational(card.getFiniteCardinality()));
+
+  Node cardUniv = nm->mkNode(kind::CARD, proxy);
+  Node leq = nm->mkNode(kind::LEQ, cardUniv, typeCardinality);
+
+  /** (=> true (<= (card (as univset t)) cardUniv) */
```

Comment:
Nit: as a general rule here, I prefer comments with `//` instead of `/** */` since the latter makes it harder to do other block comments when debugging.

Reply:
Done. What does `Nit` mean by the way?

Reply:
"nitpick", it just means its a really minor comment.